api/src/middleware/respond.ts

```diff
@@ -19,6 +21,15 @@ export const respond: RequestHandler = asyncHandler(async (req, res) => {
 
 	let exceedsMaxSize = false;
 
+	const arePermissionsCachable = await permissionsCachable(
```

Comment:
In the interest of performance [as we only ever cache on reads](https://github.com/directus/directus/blob/ea76df0d9ff8d7c3edec1ab80038e91dc373ef57/api/src/middleware/respond.ts#L29) this should only ever run on `GET` requests.

Reply:
It should really only run if all the less expensive checks have passed, so the call could be placed in the `if` in line 47.

Reply:
Should be resolved via https://github.com/directus/directus/pull/24543/commits/f254de6b3a3cefe2d8aa6239fb4a24c0d676abd0